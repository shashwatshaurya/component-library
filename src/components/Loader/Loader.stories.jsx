import React from 'react';
import Loader from './index';

export default {
  title: 'Components/Loader',
  component: Loader,
  tags: ['autodocs'],
  argTypes: {
    size: {
      control: 'select',
      options: ['sm', 'md', 'lg'],
      description: 'Size of the loader',
    },
    color: {
      control: 'select',
      options: ['primary', 'secondary', 'success', 'error'],
      description: 'Color of the loader',
    },
  },
};

const Template = (args) => <Loader {...args} />;

export const Default = Template.bind({});
Default.args = {
  size: 'md',
  color: 'primary',
};

export const Small = Template.bind({});
Small.args = {
  size: 'sm',
  color: 'primary',
};

export const Large = Template.bind({});
Large.args = {
  size: 'lg',
  color: 'primary',
};

export const CustomColor = Template.bind({});
CustomColor.args = {
  size: 'md',
  color: 'secondary',
};

export const LoaderGroup = () => (
  <div className="flex space-x-4">
    <Loader size="sm" />
    <Loader size="md" />
    <Loader size="lg" />
  </div>
);
