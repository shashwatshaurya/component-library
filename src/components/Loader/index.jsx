import React from 'react';
import PropTypes from 'prop-types';

const Loader = ({ size = 'md', color = 'primary' }) => {
  const sizes = {
    sm: 'w-4 h-4',
    md: 'w-8 h-8',
    lg: 'w-12 h-12',
  };

  return (
    <div
      role="status"
      aria-live="polite"
      className="inline-flex items-center justify-center"
    >
      <div
        className={`
          animate-spin rounded-full border-4 
          border-gray-200
          border-t-primary
          ${sizes[size]}
        `}
      />
      <span className="sr-only" role="status">
        Loading...
      </span>
    </div>
  );
};

Loader.propTypes = {
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
  color: PropTypes.string,
};

export default Loader;
