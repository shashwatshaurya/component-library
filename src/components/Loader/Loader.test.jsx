import React from 'react';
import { render, screen } from '@testing-library/react';
import Loader from './index';

describe('Loader', () => {
  it('renders loader component', () => {
    render(<Loader />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  it('has correct role for accessibility', () => {
    render(<Loader />);
    expect(screen.getByText('Loading...')).toHaveAttribute('role', 'status');
  });

  it('matches snapshot', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
