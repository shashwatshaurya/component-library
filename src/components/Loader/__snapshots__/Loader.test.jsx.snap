// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<div
  aria-live="polite"
  class="inline-flex items-center justify-center"
  role="status"
>
  <div
    class="
          animate-spin rounded-full border-4 
          border-gray-200
          border-t-primary
          w-8 h-8
        "
  />
  <span
    class="sr-only"
    role="status"
  >
    Loading...
  </span>
</div>
`;
